pkg/cmd/roachtest/run.go

```diff
@@ -327,3 +333,104 @@ func testRunnerLogger(
 	shout(ctx, l, os.Stdout, "test runner logs in: %s", runnerLogPath)
 	return l, teeOpt
 }
+
+// runOperations sequentially runs operations matched by the passed-in filter.
+func runOperations(register func(registry.Registry), filter string) error {
+	//lint:ignore SA1019 deprecated
+	rand.Seed(roachtestflags.GlobalSeed)
+	r := makeTestRegistry()
+
+	register(&r)
+	ctx := context.Background()
+	stopper := stop.NewStopper()
+	defer stopper.Stop(ctx)
+	// NB: root logger with no path always tees to Stdout.
+	l, err := logger.RootLogger("", logger.NoTee)
+	if err != nil {
+		return err
+	}
+	if len(roachtestflags.ClusterNames) == 0 {
+		return errors.New("roachtest: run-operation must be called with --cluster")
+	}
+	clusterName := roachtestflags.ClusterNames
+	// TODO(bilal): This is excessive for just getting the number of nodes in the
+	// cluster. We should expose a roachprod.Nodes method or so.
+	nodes, err := roachprod.PgURL(ctx, l, clusterName, roachtestflags.CertsDir, roachprod.PGURLOptions{})
+	if err != nil {
+		return errors.Wrap(err, "roachtest: run-operation: error when getting number of nodes")
+	}
+
+	c, err := attachToExistingCluster(ctx, clusterName, l, spec.ClusterSpec{NodeCount: len(nodes)}, attachOpt{
```

Comment:
Moving to just creating clusterImpl here for now, just to avoid the attachToExistingCluster gotchas. Thanks!

The reuse of clusterImpl is also a good point worth bringing up and is something @renatolabs and I discussed a fair bit in person last week. Sticking to `clusterImpl` has a lot of benefits as it's very familiar to anyone who has written roachtests. However the strong tie-in with the roachtest package (and the fact that attachToExistingClusters can't be called outside the roachtest package as the roachtest package can't be imported, but its subpackages can be) also has implications for `drt-run` and basically means `drt-run` will want to call `run-operation` as a subprocess instead of my preferred approach of calling operations directly.

We decided to stick with just using `clusterImpl` for now but I'd want to consider a simpler cluster interface for operations, one that disallows certain methods eg. Wipe, and then we can consider having an exported, simpler cluster implementation that's shared by the roachtest clusterImpl and the operation clusterImpl. However that's definitely further down the list of things I want to do.